src/instructlab/configuration.py

```diff
@@ -127,6 +127,74 @@ def after_debug_level(self):
         return self
 
 
+class _document_store(BaseModel):
+    """Class describing configuration of document store backend for RAG."""
+
+    uri: str = Field(
+        default=DEFAULTS.DEFAULT_DOCUMENT_STORE_PATH,
+        description="Document store service URI."
+    )
+    collection_name: str = Field(
+        default=DEFAULTS.DOCUMENT_STORE_COLLECTION_NAME,
+        description="Document store collection name."
+    )
+
+
+class _embedding_model(BaseModel):
+    """Class describing configuration of embedding parameters for RAG."""
+
+    # model configuration
+    model_config = ConfigDict(extra="ignore", protected_namespaces=())
+
+    model_dir: str = Field(
+        default_factory=lambda: DEFAULTS.MODELS_DIR,
+        description="The default system model location store, located in the data directory.",
+    )
+    model_name: str = Field(
+        default_factory=lambda: DEFAULTS.DEFAULT_EMBEDDING_MODEL,
+        description="Embedding model to use for RAG.",
+    )
+
+    def local_model_path(self) -> str:
+        if self.model_dir is None:
+            click.secho(f"Missing value for field model_dir in {vars(self)}")
+            raise click.exceptions.Exit(1)
+
+        if self.model_name is None:
+            click.secho(f"Missing value for field model_name in {vars(self)}")
+            raise click.exceptions.Exit(1)
+
+        return os.path.join(self.model_dir, self.model_name)
+
+
+class _retriever(BaseModel):
+    """Class describing configuration of retrieval parameters for RAG."""
+
+    top_k: int = Field(
+        default=DEFAULTS.RETRIEVER_TOP_K,
+        description="The maximum number of documents to retrieve."
+    )
+    embedding_model: _embedding_model = Field(
+        default=_embedding_model(),
```

Comment:
use `default_factory=_document_store` and `default_factory=_embedding_model` instead and then run `make regenerate-testdata`